content/blog/testthat-3-1/index.Rmd

```diff
@@ -0,0 +1,157 @@
+---
+output: hugodown::hugo_document
+
+slug: testthat-3-1
+title: testthat 3.1.0
+date: 2021-10-01
+author: Hadley Wickham
+description: >
+    testthat 3.1.0 brings improvements to snapshot testing and one breaking
+    change.
+
+photo:
+  url: https://unsplash.com/photos/vasU4-TlC5I
+  author: Ethan Hoover
+
+# one of: "deep-dive", "learn", "package", "programming", or "other"
+categories: [package] 
+tags: [testthat]
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+
+We're stoked to announce the release of [testthat](http://testthat.r-lib.org/) 3.1.0.
+testthat makes it easy to turn your existing informal tests into formal, automated tests that you can rerun quickly and easily.
+testthat is the most popular unit-testing package for R, and is used by over 6,000 CRAN and Bioconductor packages.
+You can learn more about unit testing at <https://r-pkgs.org/tests.html>.
+
+You can install testthat from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("{package}")
+```
+
+This release of testthat includes a bunch of minor improvements to snapshotting as well as one breaking change (which only applies if you're using the 3rd edition).
+You can see a full list of changes in the [release notes](https://github.com/r-lib/testthat/blob/master/NEWS.md).
+
+```{r setup}
+library(testthat)
+```
+
+## Snapshot tests
+
+Most of the effort in this release has gone in [snapshot tests](https://testthat.r-lib.org/articles/snapshotting.html), a new feature in testthat 3.0.0.
+snapshot tests (also known as [golden tests](https://ro-che.info/articles/2017-12-04-golden-tests)) record expect output in a separate human-readable file Instead of using code to describe expected output.
```

Comment:
```suggestion
snapshot tests (also known as [golden tests](https://ro-che.info/articles/2017-12-04-golden-tests)) record expected output in a separate human-readable file instead of using code to describe what the expected output looks like.
```